### ISSUE TITLE
Add support for loggers in LightningCLI

### ISSUE THREAD

**Issue by gatoniel** [PyTorchLightning/pytorch-lightning#7192]
## 🚀 Feature
Hi,
add support for loggers in `LightningCLI` similar to the callback support.

### Motivation
I want to use the `LightningCLI` as I want to use config files. However, I do not know how to define loggers with `LightningCLI`. The docs mention how to add callbacks in the config files. I just assumed to add the logger similarly. But this does not work. I added this to the trainer part of the config file:
```yaml
  logger:
    class_path: pytorch_lightning.loggers.TensorBoardLogger
    init_args:
      save_dir: data/tensorboard
```
But I get this error
> error: 'Configuration check failed :: No action for key "trainer.logger" to check its value.'

### Pitch
It would be nice if one could define the logger via the config file as mentioned above.

### Alternatives
It would also be possible to just pass the logger to `LightningCLI`:
```python
from pytorch_lightning.utilities.cli import LightningCLI
from pytorch_lightning import loggers as pl_loggers
tb_logger = pl_loggers.TensorBoardLogger("path/to/logs")

cli = LightningCLI(Model, logger=tb_logger)
```
But here the ability of configuration via files is lost again.

**Comment by gatoniel** [comment]
Hi,
I think this is actually not an issue for PyTorchLightning, but for jsonargparse. Somehow the combination of `Union` and `Iterable` is not working properly (https://github.com/omni-us/jsonargparse/issues/53).

In the `Trainer.__init__` the typehint for the `logger` has such a combination
https://github.com/PyTorchLightning/pytorch-lightning/blob/f58865aada46afcf374a3aca965292635e96319f/pytorch_lightning/trainer/trainer.py#L89
whereas the typehint for the `callbacks` works with `List` instead of `Iterable`
https://github.com/PyTorchLightning/pytorch-lightning/blob/f58865aada46afcf374a3aca965292635e96319f/pytorch_lightning/trainer/trainer.py#L91

I just changed `Iterable` -> `List` in my local version, i.e.
```python
        logger: Union[LightningLoggerBase, List[LightningLoggerBase], bool] = True,
```
Then, parsing of the logger via the config file worked. However, I do not know if the typehint `Iterable` is necessary here or could be exchanged with `List`.

**Comment by gatoniel** [comment]
This was only a problem for python>=3.7 and is fixed by jsonargparse v3.10.1.
